src/python/bot/tasks/task_creation.py

```diff
@@ -285,6 +285,35 @@ def request_bisection(testcase_id):
     testcase.set_metadata('requested_fixed_bisect', True)
 
 
+def _check_commits(testcase, bisect_type, old_commit, new_commit):
+  """Check old and new commit validity."""
+  if old_commit != new_commit:
+    return old_commit, new_commit
+
+  # Something went wrong during bisection for the same commit to be chosen for
+  # both the start and end range.
+  # Get the bisection infrastructure to re-bisect.
+  bucket_path = build_manager.get_primary_bucket_path()
```

Comment:
What if this is a custom binary job, do we need any bailout for that ?

Reply:
Added a check for this.